jabgui/src/main/java/org/jabref/gui/importer/fetcher/WebSearchPaneViewModel.java

```diff
@@ -85,18 +87,31 @@ public WebSearchPaneViewModel(GuiPreferences preferences, DialogService dialogSe
                     }
 
                     try {
-                        parser.parse(queryText, NO_EXPLICIT_FIELD);
+                        SearchLexer lexer = new SearchLexer(CharStreams.fromString(queryText));
+                        lexer.removeErrorListeners(); // no infos on file system
+                        lexer.addErrorListener(ThrowingErrorListener.INSTANCE);
+                        SearchParser parser = new SearchParser(new CommonTokenStream(lexer));
+                        parser.removeErrorListeners(); // no infos on file system
+                        parser.addErrorListener(ThrowingErrorListener.INSTANCE);
+                        parser.setErrorHandler(new BailErrorStrategy()); // ParseCancellationException on parse errors
+                        parser.start();
                         return null;
-                    } catch (ParseException e) {
-                        String element = e.currentToken.image;
-                        int position = e.currentToken.beginColumn;
-                        if (element == null) {
-                            return ValidationMessage.error(Localization.lang("Invalid query. Check position %0.", position));
-                        } else {
-                            return ValidationMessage.error(Localization.lang("Invalid query element '%0' at position %1", element, position));
+                    } catch (ParseCancellationException e) {
+                        // RecognitionException can point out the exact error
+                        if (e.getCause() instanceof RecognitionException) {
+                            RecognitionException recEx = (RecognitionException) e.getCause();
+                            Token offendingToken = recEx.getOffendingToken();
+
+                            // The character position is 0-based, so we add 1 for user-friendliness.
+                            int line = offendingToken.getLine();
+                            int charPositionInLine = offendingToken.getCharPositionInLine() + 1;
+
+                            String errorMessage = "Invalid syntax at line %d, position %d".formatted(line, charPositionInLine);
+                            return ValidationMessage.error(Localization.lang(errorMessage));
```

Comment:
Please reuse the existing l10n:  Localization.lang("Invalid query element '%0' at position %1,line, charPositioninLIne)...